src/main/scala/mesosphere/marathon/core/launchqueue/impl/AppTaskLauncherActor.scala

```diff
@@ -334,44 +335,56 @@ private class AppTaskLauncherActor(
       sender ! MatchedTaskOps(offer.getId, Seq.empty)
 
     case ActorOfferMatcher.MatchOffer(deadline, offer) =>
-      val newTaskOpt: Option[CreatedTask] = taskFactory.newTask(app, offer, tasksMap.values)
-      newTaskOpt match {
-        case Some(CreatedTask(mesosTask, marathonTask)) =>
-          def updateActorState(): Unit = {
-            val taskId = marathonTask.taskId
-            assume(
-              taskId.mesosTaskId == mesosTask.getTaskId,
-              "marathon task id and mesos task id must be equal"
-            )
-
-            tasksMap += taskId -> marathonTask
-            inFlightTaskLaunches += taskId -> None
-            tasksToLaunch -= 1
-            OfferMatcherRegistration.manageOfferMatcherStatus()
-          }
+      def updateActorState(task: Task, mesosTask: Option[TaskInfo]): Unit = {
+        val taskId = task.taskId
+
+        mesosTask.foreach { taskInfo =>
+          assume(
+            taskId.mesosTaskId == taskInfo.getTaskId, "marathon task id and mesos task id must be equal")
+
+          // only decrement for launched tasks, not for reservations:
+          // FIXME (217): make sure this doesn't break the counting logic
+          tasksToLaunch -= 1
+        }
+
+        tasksMap += taskId -> task
+        inFlightTaskLaunches += taskId -> None
+        OfferMatcherRegistration.manageOfferMatcherStatus()
+      }
 
-          updateActorState()
+      val taskOp: Option[TaskOp] = taskOpLogic.inferTaskOp(app, offer, tasksMap.values)
+      taskOp match {
+        case Some(launch @ TaskOp.Launch(mesosTask, newTask, _, _)) =>
+          updateActorState(newTask, Some(mesosTask))
 
           log.info("Request to launch task with id '{}', version '{}'. {}",
             mesosTask.getTaskId.getValue, app.version, status)
 
-          val launch = OfferMatcher.Launch(mesosTask, marathonTask)
-          self ! AppTaskLauncherActor.ScheduleTaskLaunchNotificationTimeout(launch)
+          self ! AppTaskLauncherActor.ScheduleTaskOpNotificationTimeout(launch)
           sender() ! MatchedTaskOps(offer.getId, Seq(TaskOpWithSource(myselfAsLaunchSource, launch)))
 
+        case Some(reserveAndCreate @ TaskOp.ReserveAndCreateVolumes(newTask, _, _, _, _)) =>
```

Comment:
Makes sense, now only matching against the type since most of the vals are not of interest.